Integrating with Mayhem

Jdkohls opened the pull request:
Pull request to submit changes for Mayhem heroes

xansec commented on go.mod:
```diff
@@ -14,6 +14,9 @@ require (
 	github.com/atotto/clipboard v0.1.4 // indirect
```
Can you explain why this file is modified? You should keep fuzzing module requirements to fuzzing targets only.

xansec commented on go.sum:
```diff
@@ -1,3 +1,4 @@
+github.com/Julusian/godocdown v0.0.0-20170816220326-6d19f8ff2df8/go.mod h1:INZr5t32rG59/5xeltqoCJoNY7e5x/3xoY9WSWVWg74=
```
Same here, any reason why this file is modified? This is auto-generated so may be a mistaken commit.

xansec commented on mayhem/Dockerfile:
```diff
@@ -0,0 +1,30 @@
+FROM fuzzers/go-fuzz:1.2.0
```
Any reason we update go here? Setting an entrypoint isn't strictly necessary. You can remove the commented commands.

Jdkohls replied:
go is updated to the most recent version (1.20.3) because fuzzers/go-fuzz:1.2.0 uses go version 1.13; the file `enola/enola.go` imports `_ "embed"`, but the `_` is not supported by go 1.13- Unfortunately, what im trying to fuzz are functions of a struct contained within this file, so updating go within the docker image seemed the easiest way around this

xansec commented on mayhem/Mayhemfile:
```diff
@@ -0,0 +1,4 @@
+project: enola-go-all
```
Targets should not contain spaces.

xansec reviewed (changes requested):
It looks like there is a stray binary in the PR as well. Please make the requested changes.

Jdkohls commented:
Edits made- I believe I reverted the go.mod and go.sum in the enola package, and added my own in the fuzzing. My justification for updating the version of go in the docker package is there, and I removed the entrypoint from docker and the space from the target in the mayhemfile.

xansec commented on fuzzing/fuzz.go:
```diff
@@ -0,0 +1,26 @@
+package fuzz
```
Minor nitpick - please rename this file (and I suppose the output binary) to represent what is being fuzzed (for example, maybe `fuzz_site_check.go`?)

xansec commented on mayhem/Dockerfile:
```diff
@@ -0,0 +1,26 @@
+FROM fuzzers/go-fuzz:1.2.0
```
Could you please add a package stage here?

Jdkohls replied:
What do you mean by this?

xansec replied:
Something like this for example
https://github.com/bmwtsn098/switchboard/blob/afc763cdc3a1623c5f384fd7ee878a2c3c7ebcdf/mayhem/Dockerfile#L20-L23
Package stages copy the resulting binary artifacts to a clean image (say ubuntu), and install only runtime dependencies (if needed) and nothing else. This helps keep image size down.

xansec reviewed (changes requested):
Please make the requested changes.

Jdkohls commented:
I believe I've done the updates and added a package stage to the dockerfile.

xansec reviewed (changes requested):
Mostly good, but you still need to restore the Dockerfile from upstream, and avoid modifying the root-level go.sum (since your fuzzing dir go.sum is already there). Please make the requested changes.

Jdkohls commented:
Added docker file, triple checked that the root-level go.sum is unmodified. It really shouldn't appear as modified, all checks that I've done have said their the same.
